pkg/controller/replicaset/replica_set_utils.go

```diff
@@ -41,7 +41,7 @@ func updateReplicaSetStatus(c appsclient.ReplicaSetInterface, rs *apps.ReplicaSe
 		rs.Status.FullyLabeledReplicas == newStatus.FullyLabeledReplicas &&
 		rs.Status.ReadyReplicas == newStatus.ReadyReplicas &&
 		rs.Status.AvailableReplicas == newStatus.AvailableReplicas &&
-		rs.Generation == rs.Status.ObservedGeneration &&
+		rs.Generation <= rs.Status.ObservedGeneration &&
```

Comment:
under what circumstances can this occur? if we get in this state, that means the controller will never reconcile to the spec unless it is updated further, which doesn't seem proper

Reply:
In fact I also don't know how can this occur like [daemon_controller.go#L1074](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemon_controller.go#L1074), but think that's correct logic.

Reply:
xref https://github.com/kubernetes/kubernetes/pull/64379#issuecomment-392446869 still remains unanswered. 

/cc @sttts @kargakis

Reply:
Given that spec and status are stored together, I can't see how ObservedGeneration could be greater than Generation except by user error (manual intervention) or programming error (bug in the controller). If one of those errors occurs, it seems reasonable for a controller that just reconciled after seeing a spec at Generation N to overwrite `ObservedGeneration = N` to more accurately represent reality.

In other words, one could argue that DaemonSet should be changed to match ReplicaSet instead, unless someone knows of a justification for DaemonSet's behavior.

Reply:
Compare with equal may be more accurate also in other controllers.

Reply:
Seems there is agreement in not moving forward with this

/close